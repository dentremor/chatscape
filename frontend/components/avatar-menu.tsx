import {
  Avatar,
  Dropdown,
  DropdownItem,
  DropdownPosition,
  DropdownToggle,
} from "@patternfly/react-core";

export const AvatarMenu = ({
  right,
  toggleAccountActions,
  accountActionsOpen,
  avatarURL,
  logOut,
  openAboutModal,
}: {
  right?: boolean;
  toggleAccountActions: () => void;
  accountActionsOpen: boolean;
  avatarURL: string;
  logOut: () => void;
  openAboutModal: () => void;
}) => (
  <Dropdown
    position={right ? DropdownPosition.right : undefined}
    onSelect={() => toggleAccountActions()}
    className="pf-u-display-flex"
    toggle={
      <DropdownToggle
        toggleIndicator={null}
        onToggle={() => toggleAccountActions()}
        aria-label="Toggle account actions"
        className="pf-u-p-0 pf-x-account-actions pf-x-avatar"
      >
        <Avatar src={avatarURL} alt="avatar" />
      </DropdownToggle>
    }
    isOpen={accountActionsOpen}
    isPlain
    dropdownItems={[
      <DropdownItem key="1" component="button" onClick={logOut}>
        Logout
      </DropdownItem>,
      <DropdownItem key="2" component="button" onClick={openAboutModal}>
        About
      </DropdownItem>,
    ]}
  />
);
