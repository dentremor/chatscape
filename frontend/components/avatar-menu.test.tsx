import { render } from "@testing-library/react";
import { AvatarMenu } from "./avatar-menu";

describe("AvatarMenu", () => {
  it("renders in the closed and left state", () => {
    expect(
      render(
        <AvatarMenu
          accountActionsOpen={false}
          toggleAccountActions={jest.fn()}
          avatarURL=""
          logOut={jest.fn()}
          openAboutModal={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders in the open and left state", () => {
    expect(
      render(
        <AvatarMenu
          accountActionsOpen={true}
          toggleAccountActions={jest.fn()}
          avatarURL=""
          logOut={jest.fn()}
          openAboutModal={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders in the closed and right state", () => {
    expect(
      render(
        <AvatarMenu
          right
          accountActionsOpen={false}
          toggleAccountActions={jest.fn()}
          avatarURL=""
          logOut={jest.fn()}
          openAboutModal={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders in the open and right state", () => {
    expect(
      render(
        <AvatarMenu
          right
          accountActionsOpen={true}
          toggleAccountActions={jest.fn()}
          avatarURL=""
          logOut={jest.fn()}
          openAboutModal={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });
});
