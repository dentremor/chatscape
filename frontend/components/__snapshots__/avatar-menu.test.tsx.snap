// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarMenu renders in the closed and left state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-c-dropdown pf-u-display-flex"
        data-ouia-component-id="OUIA-Generated-Dropdown-1"
        data-ouia-component-type="PF4/Dropdown"
        data-ouia-safe="true"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Toggle account actions"
          class="pf-c-dropdown__toggle pf-m-plain pf-u-p-0 pf-x-account-actions pf-x-avatar"
          data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
          data-ouia-component-type="PF4/DropdownToggle"
          data-ouia-safe="true"
          id="pf-dropdown-toggle-id-0"
          type="button"
        >
          <span>
            <img
              alt="avatar"
              class="pf-c-avatar"
              src=""
            />
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-c-dropdown pf-u-display-flex"
      data-ouia-component-id="OUIA-Generated-Dropdown-1"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe="true"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        aria-label="Toggle account actions"
        class="pf-c-dropdown__toggle pf-m-plain pf-u-p-0 pf-x-account-actions pf-x-avatar"
        data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
        data-ouia-component-type="PF4/DropdownToggle"
        data-ouia-safe="true"
        id="pf-dropdown-toggle-id-0"
        type="button"
      >
        <span>
          <img
            alt="avatar"
            class="pf-c-avatar"
            src=""
          />
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AvatarMenu renders in the closed and right state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-c-dropdown pf-m-align-right pf-u-display-flex"
        data-ouia-component-id="OUIA-Generated-Dropdown-3"
        data-ouia-component-type="PF4/Dropdown"
        data-ouia-safe="true"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Toggle account actions"
          class="pf-c-dropdown__toggle pf-m-plain pf-u-p-0 pf-x-account-actions pf-x-avatar"
          data-ouia-component-id="OUIA-Generated-DropdownToggle-3"
          data-ouia-component-type="PF4/DropdownToggle"
          data-ouia-safe="true"
          id="pf-dropdown-toggle-id-2"
          type="button"
        >
          <span>
            <img
              alt="avatar"
              class="pf-c-avatar"
              src=""
            />
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-c-dropdown pf-m-align-right pf-u-display-flex"
      data-ouia-component-id="OUIA-Generated-Dropdown-3"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe="true"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        aria-label="Toggle account actions"
        class="pf-c-dropdown__toggle pf-m-plain pf-u-p-0 pf-x-account-actions pf-x-avatar"
        data-ouia-component-id="OUIA-Generated-DropdownToggle-3"
        data-ouia-component-type="PF4/DropdownToggle"
        data-ouia-safe="true"
        id="pf-dropdown-toggle-id-2"
        type="button"
      >
        <span>
          <img
            alt="avatar"
            class="pf-c-avatar"
            src=""
          />
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AvatarMenu renders in the open and left state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-c-dropdown pf-m-expanded pf-u-display-flex"
        data-ouia-component-id="OUIA-Generated-Dropdown-2"
        data-ouia-component-type="PF4/Dropdown"
        data-ouia-safe="true"
      >
        <button
          aria-expanded="true"
          aria-haspopup="true"
          aria-label="Toggle account actions"
          class="pf-c-dropdown__toggle pf-m-plain pf-u-p-0 pf-x-account-actions pf-x-avatar"
          data-ouia-component-id="OUIA-Generated-DropdownToggle-2"
          data-ouia-component-type="PF4/DropdownToggle"
          data-ouia-safe="true"
          id="pf-dropdown-toggle-id-1"
          type="button"
        >
          <span>
            <img
              alt="avatar"
              class="pf-c-avatar"
              src=""
            />
          </span>
        </button>
        <ul
          aria-labelledby="pf-dropdown-toggle-id-1"
          class="pf-c-dropdown__menu"
          role="menu"
        >
          <li
            role="none"
          >
            <button
              aria-disabled="false"
              class="pf-c-dropdown__menu-item"
              data-ouia-component-id="OUIA-Generated-DropdownItem-1"
              data-ouia-component-type="PF4/DropdownItem"
              data-ouia-safe="true"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Logout
            </button>
          </li>
          <li
            role="none"
          >
            <button
              aria-disabled="false"
              class="pf-c-dropdown__menu-item"
              data-ouia-component-id="OUIA-Generated-DropdownItem-2"
              data-ouia-component-type="PF4/DropdownItem"
              data-ouia-safe="true"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              About
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-c-dropdown pf-m-expanded pf-u-display-flex"
      data-ouia-component-id="OUIA-Generated-Dropdown-2"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe="true"
    >
      <button
        aria-expanded="true"
        aria-haspopup="true"
        aria-label="Toggle account actions"
        class="pf-c-dropdown__toggle pf-m-plain pf-u-p-0 pf-x-account-actions pf-x-avatar"
        data-ouia-component-id="OUIA-Generated-DropdownToggle-2"
        data-ouia-component-type="PF4/DropdownToggle"
        data-ouia-safe="true"
        id="pf-dropdown-toggle-id-1"
        type="button"
      >
        <span>
          <img
            alt="avatar"
            class="pf-c-avatar"
            src=""
          />
        </span>
      </button>
      <ul
        aria-labelledby="pf-dropdown-toggle-id-1"
        class="pf-c-dropdown__menu"
        role="menu"
      >
        <li
          role="none"
        >
          <button
            aria-disabled="false"
            class="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-1"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe="true"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Logout
          </button>
        </li>
        <li
          role="none"
        >
          <button
            aria-disabled="false"
            class="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-2"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe="true"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            About
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AvatarMenu renders in the open and right state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-c-dropdown pf-m-align-right pf-m-expanded pf-u-display-flex"
        data-ouia-component-id="OUIA-Generated-Dropdown-4"
        data-ouia-component-type="PF4/Dropdown"
        data-ouia-safe="true"
      >
        <button
          aria-expanded="true"
          aria-haspopup="true"
          aria-label="Toggle account actions"
          class="pf-c-dropdown__toggle pf-m-plain pf-u-p-0 pf-x-account-actions pf-x-avatar"
          data-ouia-component-id="OUIA-Generated-DropdownToggle-4"
          data-ouia-component-type="PF4/DropdownToggle"
          data-ouia-safe="true"
          id="pf-dropdown-toggle-id-3"
          type="button"
        >
          <span>
            <img
              alt="avatar"
              class="pf-c-avatar"
              src=""
            />
          </span>
        </button>
        <ul
          aria-labelledby="pf-dropdown-toggle-id-3"
          class="pf-c-dropdown__menu pf-m-align-right"
          role="menu"
        >
          <li
            role="none"
          >
            <button
              aria-disabled="false"
              class="pf-c-dropdown__menu-item"
              data-ouia-component-id="OUIA-Generated-DropdownItem-3"
              data-ouia-component-type="PF4/DropdownItem"
              data-ouia-safe="true"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Logout
            </button>
          </li>
          <li
            role="none"
          >
            <button
              aria-disabled="false"
              class="pf-c-dropdown__menu-item"
              data-ouia-component-id="OUIA-Generated-DropdownItem-4"
              data-ouia-component-type="PF4/DropdownItem"
              data-ouia-safe="true"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              About
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-c-dropdown pf-m-align-right pf-m-expanded pf-u-display-flex"
      data-ouia-component-id="OUIA-Generated-Dropdown-4"
      data-ouia-component-type="PF4/Dropdown"
      data-ouia-safe="true"
    >
      <button
        aria-expanded="true"
        aria-haspopup="true"
        aria-label="Toggle account actions"
        class="pf-c-dropdown__toggle pf-m-plain pf-u-p-0 pf-x-account-actions pf-x-avatar"
        data-ouia-component-id="OUIA-Generated-DropdownToggle-4"
        data-ouia-component-type="PF4/DropdownToggle"
        data-ouia-safe="true"
        id="pf-dropdown-toggle-id-3"
        type="button"
      >
        <span>
          <img
            alt="avatar"
            class="pf-c-avatar"
            src=""
          />
        </span>
      </button>
      <ul
        aria-labelledby="pf-dropdown-toggle-id-3"
        class="pf-c-dropdown__menu pf-m-align-right"
        role="menu"
      >
        <li
          role="none"
        >
          <button
            aria-disabled="false"
            class="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-3"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe="true"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Logout
          </button>
        </li>
        <li
          role="none"
        >
          <button
            aria-disabled="false"
            class="pf-c-dropdown__menu-item"
            data-ouia-component-id="OUIA-Generated-DropdownItem-4"
            data-ouia-component-type="PF4/DropdownItem"
            data-ouia-safe="true"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            About
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
